model/project_parser.go

```diff
@@ -159,7 +158,37 @@ func (pp *ParserProject) Insert() error {
 }
 
 func (pp *ParserProject) MarshalBSON() ([]byte, error) {
-	return mgobson.Marshal(pp)
+	return bson.Marshal(pp)
+}
+
+// MarshalBSON marshals the BSON and attempts to unmarshal it back to make sure
+// it is valid. It only retries when it fails at reading the BSON, not if it encountered
+// an error while marshalling.
+func (pp *ParserProject) RetryMarshalBSON(retries int) ([]byte, error) {
+	return pp.retryMarshalBSON(retries, retries)
+}
+
+func (pp *ParserProject) retryMarshalBSON(maxRetries, retries int) ([]byte, error) {
+	projBytes, err := bson.Marshal(pp)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshalling project")
+	}
+	_, err = GetProjectFromBSON(projBytes)
+	if err != nil {
+		if retries > 0 {
+			return pp.retryMarshalBSON(maxRetries, retries-1)
+		}
+		return nil, errors.Wrap(err, "unmarshalling project to verify it's integrity")
+	}
+	// TODO (TBA-IN-REVIEW): Remove this log line, potentially the whole retry if it's
```

Comment:
I didn't create the ticket yet in case the reviewer has any thoughts about it, who ever reviews this please lmk if this is good via thumbs up or comment and I'll create the ticket

Reply:
I didn't do the investigation so I don't know if this is really the problem, but given that it affects some distros more than others, my suspicion is that this is a networking problem and not a server-side problem. That being said, I don't think there's any harm in adding retries for debugging, as long as we remember to remove it if we find it's not helping.

Reply:
I'll add that to the comment of the ticket! I linked the ticket to this ticket.

I agree that it seems like a networking problem as well rather than this though. It's the only route we send binary rather than json which is interesting